### PR TITLE
Refactor `kn vsphere` plugin command structure

### LINKED ISSUE #295: Refactor `kn vsphere` plugin to add list commands
**Is your feature request related to a problem? Please describe.**
`kn vsphere` does not support list operations and certain commands, like `login` can be improved in wording/usability for the target persona.

**Describe the solution you'd like**
Allow for equivalent `kn broker [list|create|delete]` usage.

**Describe alternatives you've considered**
Keep as is.

**Additional context**

### PR DESCRIPTION
**Pull request by embano1**
Fixes #295

Added and updated existing unit/E2E tests. Command syntax now aligns with `kn` commands, e.g. `kn [resource] create`. The packages rely on several `kn` packages, e.g. for pretty-printing in `list` output.

To ease the review, the commits are broken into:

- dependencies: `vendor` and `third_party`
- build/e2e
- docs
- CLI/package related (incl. unit tests)

## Proposed Changes

- :gift: Improve `kn vsphere` CLI experience

```console
./kn-vsphere
Knative plugin to create Knative compatible Event Sources for VMware vSphere events,
and Bindings to access the vSphere API

Usage:
  kn-vsphere [command]

Available Commands:
  auth        Manage vSphere credentials
  binding     Manage vSphere API bindings
  completion  generate the autocompletion script for the specified shell
  help        Help about any command
  source      Manage vSphere Event Sources
  version     Prints the plugin version

Flags:
  -h, --help   help for kn-vsphere

Use "kn-vsphere [command] --help" for more information about a command.
```

### Pre-review Checklist

- [x] **At least 80% unit test coverage**
- [ ] **E2E tests** for any new behavior
- [x] **Docs** for any user-facing impact

**Release Note**

```release-note
The `kn vsphere` CLI plugin to manage `VSphereSources` and `VSphereBindings` was refactored to include new commands, e.g. `list` and `delete`. Also `kn vsphere login` is replaced with a new `kn vsphere auth` command to manage credentials (secrets).
Action required: update custom scripts using `kn vsphere` with the new syntax.
```

### CODE REVIEW

**Review by evankanderson — COMMENTED** [review]
About halfway through ... this got pretty big

File: plugins/vsphere/pkg/command/binding/list.go
Comment (evankanderson) [reply plugins/vsphere/pkg/command/binding/list.go:107]: (I think yo uspelled these out because getting `additionalPrinterColumns` is hard, so I'd just mention that here.

File: plugins/vsphere/pkg/command/source/create.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/create.go:41]: I'd consider using positional args for the resource target ("name").

File: plugins/vsphere/pkg/command/source/create.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/create.go:62]: Consider using the [k8s `sets.String`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String) here.

File: plugins/vsphere/pkg/command/source/create.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/create.go:96]: Maybe call this `vcenter` or `vc-url`? I confused this with `sink-uri` when reading some other code.

File: plugins/vsphere/pkg/command/source/delete.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/delete.go:33]: Again, consider using positional args

File: plugins/vsphere/pkg/command/source/list.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/list.go:61]: `v1.NamespaceAll`?

File: plugins/vsphere/pkg/command/source/list.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/list.go:124]: Why not:

```golang
for _, src := range sourceList.Items {
  r, err := printSource(&src, printOptions)
  ...
}
```

File: plugins/vsphere/pkg/command/source/source.go
Comment (evankanderson) [inline plugins/vsphere/pkg/command/source/source.go:59]: Why not just use [`apis.ParseURL` here from Knative](https://pkg.go.dev/knative.dev/pkg@v0.0.0-20211213170220-c3a4928da77b/apis#ParseURL)? I think it does _exactly_ what you want.

**Review by evankanderson — COMMENTED** [review]
Mostly suggestions on API surface to feel a bit more like `kn` and `kubectl`.

File: plugins/vsphere/pkg/command/auth/auth.go
Comment (embano1) [reply plugins/vsphere/pkg/command/auth/auth.go:38]: Certainly would be nice to have, though I'd like to defer this until someone asks :)

File: plugins/vsphere/pkg/command/source/delete.go
Comment (embano1) [reply plugins/vsphere/pkg/command/source/delete.go:33]: Agree, I will open an issue for this as it would be a bigger rewrite which I'd like to defer for now to get this in.

Comment (embano1) [reply plugins/vsphere/pkg/command/source/delete.go:33]: https://github.com/vmware-tanzu/sources-for-knative/issues/347

File: plugins/vsphere/pkg/command/source/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/source/create.go:96]: Agree, will change.

File: plugins/vsphere/pkg/command/source/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/source/create.go:41]: Agree, see earlier comment above.

File: plugins/vsphere/pkg/command/binding/list.go
Comment (embano1) [reply plugins/vsphere/pkg/command/binding/list.go:107]: I followed/borrowed this from how `kn` does it.

File: plugins/vsphere/pkg/command/binding/list.go
Comment (embano1) [reply plugins/vsphere/pkg/command/binding/list.go:61]: ack

File: plugins/vsphere/pkg/command/binding/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/binding/create.go:38]: :(

File: plugins/vsphere/pkg/command/auth/delete.go
Comment (embano1) [reply plugins/vsphere/pkg/command/auth/delete.go:47]: To make the (my) linter happy. Will revert.

File: plugins/vsphere/pkg/command/auth/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/auth/create.go:134]: I think so, yeah. Haven't looked at this code before.

File: plugins/vsphere/pkg/command/auth/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/auth/create.go:72]: Yup, I like that.

File: plugins/vsphere/pkg/command/auth/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/auth/create.go:57]: I'd rather keep it explicit for now tbh.

File: plugins/vsphere/pkg/command/auth/create.go
Comment (embano1) [reply plugins/vsphere/pkg/command/auth/create.go:48]: Dunno, ask the author :) Changed, thx.

**Comment by embano1** [comment]
Thx @evankanderson for your thorough reviewed. Addressed most of it, rest is deferred and tracked (issue).

**Comment by gabo1208** [comment]
Waiting for @evankanderson approval to merge

**Review by evankanderson — DISMISSED** [review]
This seems like an improvement; sorry I didn't make that clear in December.

**Comment by embano1** [comment]
Thx @evankanderson

> This seems like an improvement; sorry I didn't make that clear in December.

Do you want me to update the release notes or description then?

Btw: looks like there's merge conflicts now which I have to address, will work on it.

**Review by evankanderson — DISMISSED** [review]
/approve

**Review by gabo1208 — APPROVED** [review]

**Review by evankanderson — APPROVED** [review]

**Pull request merged**
